Déplacements-chargeAuto-début de combat

Nicolas62x opened the pull request:
Bon j'ai tout mis dans un seul fichier car en plus d'être plus pratique, ne bug pas, si vous tenez aux multiples fichier je vous laisse séparer mon code, enfin bref
Le robot est désormais capable de se déplacer a des coordonnées précises grace a moveTo, il se charge de façon auto sur le chemins, j'ai du modifier la fonction de charge qui n'était pas répétable (ça marchait que la première fois)  il reste donc a dev  une fonction qui s'occupe de déplacer le robot d'îles de chaos en îles  de chaos et de le combattre à chaque fois

le robot se crois en 0 0 0 regardant le nord lorsqu'on lance le programme suffis de changer le x y z et direction pour changer ça
(pour la direction 0 = east 1=south 2=west 3=north)

![2019-03-17_15 05 44](https://user-images.githubusercontent.com/48631004/54492528-bbebdf00-48c7-11e9-9255-6e76b754d5b4.png)
![2019-03-17_15 06 23](https://user-images.githubusercontent.com/48631004/54492534-bee6cf80-48c7-11e9-9025-d99ed6218163.png)

Nicolas62x committed:
Déplacements-chargeAuto-avec coordonnées

Bon j'ai tout mis dans un seul fichier car en plus d'être plus pratique, ne bug pas, si vous tenez aux multiples fichier je vous laisse séparer mon code, enfin bref
Le robot est désormais capable de se déplacer a des coordonnées précises race a moveTo, il se charge de façon auto sur le chemins, j'ai du modifier la fonction de charge qui n'était pas répétable (ça marchait que la première fois)  il reste donc a dev  une fonction qui s'occupe de déplacer le robot d'îles de chaos en îles  de chaos et de le combattre à chaque fois

NameP01NTER commented:
À ne pas merge. Cela rendrais la collaboration quasi impossible ! Si le programme a été séparé c'est qu'il y a une raison.

Nicolas62x commented:
bah je trouve plus simple comme ça je vois pas pourquoi tu tiens tant à séparé le programme mais ducoup stv tu peux séparé le miens

NameP01NTER commented:
Enfaite si tu ne sépare pas alors plusieurs personnes ne peuvent plus travailler sur le même projet, car github ne peux pas mélanger les code à l'intérieur d'un même fichier

Nicolas62x commented:
Ouai sauf que si tu sépares personne ne va proposer de vrais améliorations car c’est trop compliqué

NameP01NTER commented:
Enfaite ça ne le rend pas plus compliqué, au contraire car on connais la fonction de chaque fichier. Si tout est regroupé on s'y perd très facilement. Bien sûr, si tu n'a pas beaucoup d'expérience en POO (même si ce n'est pas vraiment de la POO) c'est compliqué pour toi.

Nicolas62x commented:
Le problème que j’ai eu c’est que lorsque je faisait une mod ça la faisait pas vraiment genre je devais reboot a chaque fois que je voulais lancer le programme et j’ai l’habitude de faire des.h et tout en c mais la je comprend pas la logique et je penses que la majorité des gens sera comme moi

Ascor8522 commented:
> Enfaite si tu ne sépare pas alors plusieurs personnes ne peuvent plus travailler sur le même projet, car github ne peux pas mélanger les code à l'intérieur d'un même fichier

Bien sur que c'est possible,
ça s'appelle un 3-way merge et on peut facilement régler les conflicts de merge.

> Enfaite ça ne le rend pas plus compliqué, au contraire car on connais la fonction de chaque fichier. Si tout est regroupé on s'y père très facilement. Bien sûr, si tu n'a pas beaucoup d'expérience en POO (même si ce n'est pas vraiment de la POO) c'est compliqué pour toi.

Sauf que Lua n'a pas ce concept de classes, et le projet est tellement petit qu'il n'y a même pas besoin de le découper en modules à mon avis. Tant que les fonctions sont bien nommées et qu'elles se limitent à 1 seule action ça devrait aller. Aussi, avec 1 seul fichier tu as déjà des possibles erreurs de modules qui ne sont plus un problème.

NameP01NTER commented:
Je propose quon laisse @Fuzay décider si il préfère le programme modulaire ou en un seul fichier. Après tout c'est sa série. Et le 3-way merge n'est vraiment pas parfait. Il oblige de faire des PR à chaque fois qu'on change une fonction car si un autre contributeur y a effectué ses modifs, ca les remplaces tous si les code sont très différents

Nicolas62x commented:
Enfin quelqu’un de logique

Nicolas62x commented:
Oui

Nicolas62x commented:
En tout cas je pense que ma version devrait être la nouvelle base puisqu’elle est plus avancée après si vous avez des améliorations à faire dessus je suis a votre écoute

MrStrikes commented:
Hello ! Tout centraliser dans un seul fichier ne ferait que rendre le tout illisible, et impossible a lire. On se retrouverait avec des fichiers bien trop longs. Le principe de module permet ainsi une meilleur lisibilité et collaboration sur le travail. Le merging est loin d'être une solution adéquate sur des projets Open Source (j'vois mal Linux sur un seul fichier énorme par exemple :/)

Nicolas62x commented:
Ceci n’est pas linux et c’est du lua, je suis d’accord qu’avec des languages plus précis il faut séparer les fichiers mais la séparation rend moins lisible justement ya pas assez de code et le lua est nul pour faire ça

MrStrikes commented:
Par Linux, je parlais du Linux Kernel, qui est un projet open source. Tout est centralisé sous modules.
Actuellement, si on suit le chemin du fichier unique, on se dirige vers du spaghetti code, ce qui est l'enfer d'un projet open source. C'est quelque chose qu'on apprends assez vite en arrivant dans le monde de l'open-source.

Nicolas62x commented:
Comme dis plus haut je suis d’accord avec toi mais toujours comme dis plus haut ce n’est pas linux ni linux kernel, je vais te faire une analogie pour que tu comprennes, pour faire de pâtes avoir une usine est le plus rentable pourtant si je te demande de faire 500g de pâtes ça m’étonnerais que tu fasse une usine

NameP01NTER commented:
Merci strikes! Heureux que quelqu'un partage mon opinion !

Nicolas62x commented:
moi aussi je la partage c'est juste que je gagne mon temps a ne pas le faire car comme dis plus haut vu la complexité du code faire une usine est une perte de temps

Nicolas62x commented:
perso je suis bloqué à savoir comment faire combattre le robot car ça s'annonce compliquer ça pour le coup, c'est pas le code qui est compliqué

MrStrikes commented:
Si on reprends ton analogie, c'est qu'on sera plusieurs a faire ces 500g de pates, donc autant s'organiser nan ?

Nicolas62x commented:
bah perso je suis pas loing d'avoir finis mais je partage au fur a mesure pour que vous puissiez m'aider mais visiblement tout ce que vous voulez faire c'est débattre sur le fait de faire un truc que je ferais pas de toute façon, la je fais la fonction pour casser toutes les tours

Nicolas62x committed:
nouvelle MAJ début de déstruction des healers

ajout du fonction déplaçant le robot a un nouveau chaos ainsi qu'un fonction cassant tout les healers externes,  mais petit problème le dragon tue le robot lorsqu'il attaque les healers et le casse
ajout aussi de quelques lignes de code pour que les blocs / entités ne déboussolent plus le robot

Wistaro commented:
Hello, je rejoins Strikes et NamePo1nter, travailler avec plusieurs modules permet d'avoir un code plus lisible, et c'est plus faciles pour les commits. 

Ou alors ouvrir plusieurs branches,  mais se serait un peu inutile et contre-productif.

Nicolas62x commented:
S’il vous plaît réfléchissez plutôt à une façon de combattre le chaos guardian, si vous voulez séparer en plusieurs modules faites le au lieu d’essayer de convaincre les autres que c’est ce qu’il faut faire

ttatanepvp123 commented:
personnellement je suis un grand partisans de la séparation de code.
c'est juste impossible de dev sans séparation.
![Capture d’écran de 2019-03-18 01-26-02](https://user-images.githubusercontent.com/35320675/54500428-f469d800-491c-11e9-8889-972a56b104a6.png)

ttatanepvp123 commented:
personnellement je suis un grand partisans de la séparation de code.
c'est juste impossible de dev sans séparation.
![Capture d’écran de 2019-03-18 01-26-02](https://user-images.githubusercontent.com/35320675/54500471-5cb8b980-491d-11e9-9a2e-dd9afd71d4c2.png)
juste le main alors imaginez sans séparation ...
![Capture d’écran de 2019-03-18 01-36-34](https://user-images.githubusercontent.com/35320675/54500586-6a227380-491e-11e9-96e2-e221a7a98431.png)

NameP01NTER commented:
Moi perso je vais intégrer les changements de Nicolas dans les différents modules et je vais ensuite refaire une PR

Fuzay commented:
On reste sur l'idée de modules ^^

Nicolas62x commented:
ok au moins c clair

Nicolas62x commented:
ducoup j'essaie de séparer les modules, ça semble fonctionné pour l'instant
![image](https://user-images.githubusercontent.com/48631004/54524746-05ddcf00-4973-11e9-99cd-0073ee1a2ca6.png)

Nicolas62x commented:
je crois que j'ai trouvé un moyen pour que le robot survive

Nicolas62x commented:
il se cache après chaque attaque

![image](https://user-images.githubusercontent.com/48631004/54525289-29ede000-4974-11e9-9a2c-f0b443e67619.png)

Nicolas62x commented:
sauf qu'il ya un problème, le temps qu'il casse le deuxième healer le premier a respawn :(

Nicolas62x committed:
Bon je sépare les fichiers
